### PR TITLE
add certificates claims+

### PR DESCRIPTION
**Pull request by bryanchriswhite**
Usage
---
```
$ certificates claims --help
Print claim information

Usage:
  /home/user/go/bin/certificates claims [flags]

Flags:
      --authorization-dburl string   url to the certificate signing authorization database (default "bolt:///home/user/.local/share/storj/cert-signing/authorizations.db")
      --raw                          if true, the raw data structures will be printed
...

Global Flags:
      --config-dir string           main directory for certificates configuration (default "/home/user/.local/share/storj/cert-signing")
...
```

Example
---
```
identity create test --difficulty 0
identity create certificates --difficulty 0
certificates setup --server.address 127.0.0.1:8888
certificates auth create 1 user@example.com
token=$(certificates auth export --all | cut -d"," -f2)
certificates run &
certs_id=$?
identity authorize test $token

certificates claim # <----
# [
#         {  
#                 "UserID": "user@example.com",
#                 "Token": "user@example.com:1JCzrVAv9MdV26ob7JLewLLXm3fYJ5grniN57bVtZYkxgMY4vsXrrYXG86MXPdtVeTGS1DfKLuPsuNBzhgQgqAFtAG4W1m",
#                 "Claim": {
#                         "Addr": "127.0.0.1:42640",
#                         "Timestamp": "2019-02-01 22:05:26 +0100 CET",
#                         "NodeID": "1263QtYjg2njUT7eEzei5snxm4buhwAUinDijRT6vCLGgJeSZAp"
#                 }          
#         }          
# ]

kill $certs_id
```

### CODE REVIEW

File: cmd/certificates/claims.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+type printableAuth struct {
+	UserID string
+	Token  string
+	Claim  *pClaim
+}
+type pClaim struct {
+	Addr      string
+	Timestamp string
```
Comment (bryanchriswhite) [inline cmd/certificates/claims.go]: Should just say `Time`

File: cmd/certificates/claims.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+)
+
+func init() {
+	rootCmd.AddCommand(claimsCmd)
+
+	cfgstruct.Bind(claimsCmd.Flags(), &claimsCfg, cfgstruct.ConfDir(defaultConfDir))
+}
+
+func cmdClaims(cmd *cobra.Command, args []string) error {
+	authDB, err := claimsCfg.NewAuthDB()
```
Comment (bryanchriswhite) [inline cmd/certificates/claims.go]: Close `authDB`

File: cmd/certificates/claims.go (hunk 1 above)
Comment (bryanchriswhite) [reply cmd/certificates/claims.go]: Yes :p

File: pkg/certificates/certificates.go [hunk 3]
```diff
@@ -305,7 +305,24 @@ func (a *AuthorizationDB) UserIDs() ([]string, error) {
 	return keys.Strings(), nil
 }
 
-// Claim marks an authorization as claimed and records claim information
+// List returns all authorizations in the database.
+func (a *AuthorizationDB) List() (auths Authorizations, _ error) {
```
Comment (navillasa) [inline pkg/certificates/certificates.go]: why the blank identifier for this error? just curious

File: cmd/certificates/claims.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	cfgstruct.Bind(claimsCmd.Flags(), &claimsCfg, cfgstruct.ConfDir(defaultConfDir))
+}
+
+func cmdClaims(cmd *cobra.Command, args []string) (err error) {
+	authDB, err := claimsCfg.NewAuthDB()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = authDB.Close()
```
Comment (egonelbre) [inline cmd/certificates/claims.go]: Use `err = errs.Combine(err, authDB.Close()`, otherwise it will override the previous error return values.

File: cmd/certificates/claims.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+	Token  string
+	Claim  *pClaim
+}
+type pClaim struct {
+	Addr   string
+	Time   string
+	NodeID string
+}
+
+func toPrintableAuth(a *certificates.Authorization) *printableAuth {
```
Comment (egonelbre) [inline cmd/certificates/claims.go]: maybe instead of `a` `auth` would be better

File: cmd/certificates/claims.go [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+	fmt.Println(string(jsonBytes))
+	return err
+}
+
+type printableAuth struct {
+	UserID string
+	Token  string
+	Claim  *pClaim
+}
+type pClaim struct {
```
Comment (egonelbre) [inline cmd/certificates/claims.go]: maybe instead of `pClaim` use `printableClaim`

File: pkg/certificates/certificates.go (hunk 3 above)
Comment (egonelbre) [inline pkg/certificates/certificates.go]: It's hard to follow code with short codes when jumping between packages https://github.com/storj/storj/wiki/Style#variable-naming. Maybe `a` -> `authdb`.

Comment (bryanchriswhite) [reply pkg/certificates/certificates.go]: @navillasa I wanted `auths` to be named so that I don't have to declare it in the method body but it's all or none with named return vars and since `err` is always to the left of `:=`, naming it would allocate an unused variable I think.

File: cmd/certificates/claims.go (hunk 4 above)
Comment (bryanchriswhite) [reply cmd/certificates/claims.go]: :+1:

File: cmd/certificates/claims.go (hunk 5 above)
Comment (bryanchriswhite) [reply cmd/certificates/claims.go]: :+1:

File: cmd/certificates/claims.go (hunk 6 above)
Comment (bryanchriswhite) [reply cmd/certificates/claims.go]: I think I planned on this but got sidetracked and never came back to it. Thanks for the reminder :p

File: pkg/certificates/certificates.go (hunk 3 above)
Comment (bryanchriswhite) [reply pkg/certificates/certificates.go]: :+1:
